### PR TITLE
add window code extraction

### PR DESCRIPTION
**Pull request by buckley-w-david**
Add snippets to extract the codes that are used by Window elements

### CODE REVIEW

File: python/src/energuide/snippets.py [hunk 1]
```diff
@@ -95,12 +95,54 @@ def _wall_code_snippet(wall_code: etree.ElementTree) -> typing.Dict[str, typing.
         'componentTypeSizeFrench': component_type_size_french,
     }
 
+def _window_code_snipped(window_code: etree.ElementTree) -> typing.Dict[str, typing.Any]:
```
Comment (JasonMWhite) [inline python/src/energuide/snippets.py]: other methods are called `snippet`

File: python/src/energuide/snippets.py [hunk 2]
```diff
@@ -95,12 +95,54 @@ def _wall_code_snippet(wall_code: etree.ElementTree) -> typing.Dict[str, typing.
...
+        'coatingsTintsEnglish': coatings_tints_english,
+        'coatingsTintsFrench': coatings_tints_french,
+        'fillTypeEnglish': fill_type_english,
+        'fillTypeFrench': fill_type_french,
+        'spacerTypeEnglish': spacer_type_english,
+        'spacerTypeFrench': spacer_type_french,
+        'typeEnglish': type_english,
+        'typeFrench': type_french,
+        'frameMaterialEnglish': frame_material_english,
+        'frameMaterialFrench': frame_material_french,
```
Comment (JasonMWhite) [inline python/src/energuide/snippets.py]: Are all of these required in the HOIS? Let's make sure to start from exactly what's necessary to replicate the information contained there. We can add to it later.

File: python/tests/test_snippets.py [hunk 3]
```diff
@@ -116,6 +116,67 @@ def test_code_snippet(code: etree.ElementTree) -> None:
...
+                'frameMaterialFrench': 'Bois',
+            }, {
+                'id': 'Code 13',
+                'label': '330204',
+                'glazingTypeEnglish': 'Triple/triple with 1 coat',
+                'glazingTypeFrench': 'Triple/triple, 1 couche',
+                'coatingsTintsEnglish': 'Low-E .20 (hard1)',
+                'coatingsTintsFrench': 'Faible E .20 (Dur 1)',
+                'fillTypeEnglish': '13 mm Air',
+                'fillTypeFrench': '13 mm d\'air',
```
Comment (JasonMWhite) [inline python/tests/test_snippets.py]: Prefer using `""` quotes instead of escaping a `\'`

File: python/src/energuide/snippets.py (hunk 2 above)
Comment (buckley-w-david) [reply python/src/energuide/snippets.py]: They're all required, the HOIS has a `TYPE` column that is a combination of all six of these fields

File: python/tests/test_snippets.py [hunk 4]
```diff
@@ -116,6 +116,67 @@ def test_code_snippet(code: etree.ElementTree) -> None:
                 'structureTypeFrench': 'Ossature de bois',
                 'componentTypeSizeEnglish': '38x89 mm (2x4 in)',
                 'componentTypeSizeFrench': '38x89 (2x4)',
-            }]
+            }],
+            'window': [{
```
Comment (JasonMWhite) [inline python/tests/test_snippets.py]: This test is getting a bit long. Maybe break it up into separate tests, and not necessarily testing the whole output.

File: python/src/energuide/snippets.py (hunk 1 above)
Comment (buckley-w-david) [reply python/src/energuide/snippets.py]: Slip of the fingers

File: python/src/energuide/snippets.py (hunk 2 above)
Comment (JasonMWhite) [reply python/src/energuide/snippets.py]: 👍

**Pull request merged**